padertorch/data/fragmenter.py

```diff
@@ -109,86 +119,145 @@ def fragment(key, x):
-    array([ 1,  4,  7, 10, 13])
-    >>> get_segment_offsets(24, 10, 3, offset_mode='center', num_segments=2)
-    array([1, 4])
-    >>> get_segment_offsets(24, 10, 3, offset_mode='center', num_segments=1)
-    array([1])
+        2xB numpy array with start and end values for B boundaries
+
+    >>> np.random.seed(0)
+    >>> get_segment_boundaries(25, 10, None, mode='begin')
+    array([[ 0, 10],
+           [10, 20]])
+    >>> get_segment_boundaries(25, 10, 3, mode='begin')
+    array([[ 0,  3,  6,  9, 12, 15],
+           [10, 13, 16, 19, 22, 25]])
+    >>> get_segment_boundaries(24, 10, 3, mode='begin')
+    array([[ 0,  3,  6,  9, 12],
+           [10, 13, 16, 19, 22]])
+    >>> get_segment_boundaries(24, 10, 3, mode='random')
+    array([[ 0,  3,  6,  9, 12],
+           [10, 13, 16, 19, 22]])
+    >>> get_segment_boundaries(24, 10, 3, mode='random_max_segments')
+    array([[ 1,  4,  7, 10, 13],
+           [11, 14, 17, 20, 23]])
+    >>> get_segment_boundaries(24, 10, 3, mode='center')
+    array([[ 1,  4,  7, 10, 13],
+           [11, 14, 17, 20, 23]])
+    >>> get_segment_boundaries(24, 10, 3, mode='center', choose_one=True)
+    array([[ 1],
+           [11]])
     """
     assert num_samples >= length, (num_samples, length)
     if shift is None:
         shift = length
     assert shift > 0, shift
 
-    if offset_mode == 'start':
+    if mode == 'begin':
         start = 0
-    elif offset_mode == 'random':
-        start = np.random.randint(num_samples - length)
-    elif offset_mode in ['end', 'center']:
+    elif mode == 'random':
+        start = rng.randint(shift)
+    elif mode == 'random_max_segments':
+        start = rng.randint((num_samples - length) % shift)
+    elif mode in ['end', 'center']:
         remainder = (num_samples - length) % shift
-        if offset_mode == 'end':
+        if mode == 'end':
             start = remainder
         else:
             start = remainder // 2
     else:
-        raise ValueError('Unknown offset mode', offset_mode)
-    offset = np.arange(start, num_samples - length + 1, shift)
-
-    if num_segments:
-        max_segments = (num_samples - length) // shift + 1
-        assert num_segments <= max_segments, (num_segments, max_segments)
-        offset = offset[:num_segments]
-    return offset
+        raise ValueError('Unknown offset mode', mode,
+                         'choose on of', possible_boundary_modes)
+    start = np.arange(start, num_samples - length + 1, shift)
+    if choose_one:
+        start = rng.choice(start, 1)
+    stop = start + length
+    boundaries = np.stack([start, stop], axis=0)
+    return boundaries
 
 
-def segment(x, length, offset=None, shift=None, axis=-1,
-            offset_mode='start', num_segments=None):
+def segment(x, axis=-1, boundary=None, length: int=None, shift=None,
+            mode='begin', choose_one=False, rng=np.random):
     """
+    Segments a signal x along an axis. Either with predefined segment
```

Comment:
Use backticks when referencing arguments in the description text. This looks way nicer when compiled to HTML.

Reply:
done